StoDevX/stograde
# More tests

- **Fix bug where test results are not added to output file**
- Test coverage to 100%
- Remove bidict dependency
- Reincorporate html formatter
- Update gist functionality
- Fixes #70 

More detailed:
- Rename `check_submit_date.py` to `get_modification_time.py`
- Don't record "not a git repository" as an unmerged branch
- Add `FormatType` to represent if Markdown or HTML is used
- Add `FormattedResult` to represent a formatted assignment
- Add `GroupType` to represent how the files should be grouped
- Update html formatter
  - Add free template to make it look nice
  - Add anchors to make jumping to a specific student possible
- Create `spec_repos.py`
  - Removes need for bidict dependency
  - Allows both ssh and https urls to be used to determine course
  - Creates a list of supported courses
- Move `get_students` to separate file
- Use GitHub personal access token to authenticate gist instead of password
  - Using a password will be deprecated in November 2020 according to https://docs.github.com/en/rest/overview/other-authentication-methods#via-username-and-password
